Allow leader to remove collect results without an explicit DELETE. (#292)
ietf-wg-ppm/draft-ietf-ppm-dap: draft-ietf-ppm-dap.md

GET requests to the collect job URI with an HTTP error status and a problem
document as described in {{errors}}.

The collector may send an HTTP DELETE request to the collect job URI, to which
the leader MUST respond with HTTP status 204 No Content. The leader MAY respond

> with HTTP status 204 No Content for requests to a collect job URI which has not

Reviewer: I think it would be nice to add a little "(e.g. because the data has been deleted due to age or lack of space)". It's a little confusing if you don't have that on your mind.

GET requests to the collect job URI with an HTTP error status and a problem
document as described in {{errors}}.

> The leader MUST remove a collect job's results when the collector sends an HTTP
> DELETE request to the collect job URI. The leader responds with HTTP status 204
> No Content for requests to a collect job URI whose results have been removed.

Reviewer 2: I think this MUST should be rewritten to describe externally observable behavior. For instance:

```suggestion
The collector may send an HTTP DELETE request to the collect job URI, to which
the leader MUST respond with HTTP status 204 No Content. The leader MUST
respond to subsequent requests to the collect job URI with HTTP status 204 No
Content.
```

Note that this also specifies how the leader should respond to the DELETE, which is missing in the existing text. My intuition is that there's no way for other protocol participants to know whether or not the leader's storage still has some aggregate share. The only thing they can see is how the leader responds to a request.

Besides that: this text makes it clear what a leader MUST do if it gets an explicit DELETE, but I think we also need a MAY sentence making it clear that the leader can start respond to the collect URI with HTTP 204 _before_ a collector sends DELETE.

Author: OK -- I had left the `MAY` off since it's IMO implied by the updated text, but explicit is usually better than implicit.

GET requests to the collect job URI with an HTTP error status and a problem
document as described in {{errors}}.

> The collector may send an HTTP DELETE request to the collect job URI, to which

Reviewer 3: ```suggestion
The collector can send an HTTP DELETE request to the collect job URI, to which
```
